contrib/cirrus/lib.sh

```diff
@@ -288,3 +288,40 @@ remove_packaged_podman_files() {
     # Be super extra sure and careful vs performant and completely safe
     sync && echo 3 > /proc/sys/vm/drop_caches || true
 }
+
+# Execute make localbenchmarks in $CIRRUS_WORKING_DIR/data
+# for preserving as a task artifact.
+localbenchmarks() {
+    local datadir
+    req_env_vars DISTRO_NV PODBIN_NAME PRIV_NAME TEST_ENVIRON TEST_FLAVOR
+    req_env_vars VM_IMAGE_NAME EC2_INST_TYPE
+
+    datadir=$CIRRUS_WORKING_DIR/data
+    mkdir -p $datadir
+
+    # Most of these are defined by cirrus and it's YAML.
+    # shellcheck disable=SC2154,SC2153
+    cat <<EOF>$datadir/benchmarks.env
+CIRRUS_BASE_BRANCH=$CIRRUS_BASE_BRANCH
+CIRRUS_BRANCH=$CIRRUS_BRANCH
+CIRRUS_BUILD_ID=$CIRRUS_BUILD_ID
+CIRRUS_CHANGE_IN_REPO=$CIRRUS_CHANGE_IN_REPO
+CIRRUS_CRON=$CIRRUS_CRON
+CIRRUS_PR=$CIRRUS_PR
+CIRRUS_TASK_ID=$CIRRUS_TASK_ID
+DISTRO_NV=$DISTRO_NV
+PODBIN_NAME=$PODBIN_NAME
+PRIV_NAME=$PRIV_NAME
+TEST_ENVIRON=$TEST_ENVIRON
+TEST_FLAVOR=$TEST_FLAVOR
+VM_IMAGE_NAME=$VM_IMAGE_NAME
+INST_TYPE=$EC2_INST_TYPE
+UNAME_RM=$(uname -r -m)
+CPUTOTAL=$(echo $(($(egrep 'processor[[:space:]]+:' /proc/cpuinfo | tr -d '[:blank:]' | cut -d: -f2 | sort -n | tail -1) + 1)) )
```

Comment:
Does this really have to be so complicated? `grep -ce '^processor' /proc/cpuinfo` is instantly grokkable, no brain overheating needed.

Also, these don't need to be spelled out as CPUTOTAL MEMTOTAL ETC. It might be nicer for readers to just have them spelled out: `cpu total:`, `Mem Total`

Reply:
Oh, `grep -c` will you look at that!  Didn't know that existed, thanks!

> These don't need to be spelled out as

This file is intended to be machine-readable metadata details indicating the basis for the benchmarks.  It's not needed now, but immensely useful if somebody wants to do any trend analysis or averaging over long periods of time.  Then, they will need a machine-readable way to say "These, ran under the exact same conditions, as those [benchmarks]".

I can add a comment-line at the top of the file so there's no confusion.